rust/src/binaryview.rs

```diff
@@ -667,6 +668,119 @@ pub trait BinaryViewExt: BinaryViewBase {
             )
         };
     }
+
+    /// Creates a new [TagType] and adds it to the view.
+    ///
+    /// # Arguments
+    /// * `name` - the name for the tag
+    /// * `icon` - the icon (recommended 1 emoji or 2 chars) for the tag
+    fn create_tag_type<N: BnStrCompatible, I: BnStrCompatible>(
+        &mut self,
```

Comment:
Remove all instances of `&mut self` in this PR; Nothing _in Rust_ is being mutated, so it shouldn't matter from a Rust perspective that you're mutating data across an FFI boundary.   Said another way, `mut` in Rust does not mean the opposite of `const` in C++, we're not adding safety rails, we're describing dataflow...within a couple ballparks.